openlibrary/catalog/get_ia.py

```diff
@@ -26,16 +26,17 @@ class NoMARCXML(IOError):
     pass
 
 
-def urlopen_keep_trying(url):
+# TODO: there are few other modules that call this, refactor to rename later to request_keep_trying
+def urlopen_keep_trying(url, headers=None):
+    """Tries to request the url three times, raises HTTPError if 403, 404, or 416.  Returns a requests.Response"""
     for i in range(3):
         try:
-            f = urllib.request.urlopen(url)
-            return f
```

Comment:
f is a https://docs.python.org/3/library/http.client.html#http.client.HTTPResponse

Reply:
Yes, understood.